spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.describe ArticlesController, type: :controller do
+
+  describe 'GET #new' do
+    context 'when user is logged in' do
+
+      let(:user) { create(:user) }
+
+      before(:each) do
+        allow(controller).to receive(:current_user).and_return(user)
+      end
+
+      it 'responds with a 200 OK HTTP response' do
+        get :new
+        expect(response.status).to eq(200)
+      end
+
+
+      it 'renders the new template' do
+        get :new
+        expect(response).to render_template "new"
+      end
+    end
+
+    context 'when user is not logged in' do
+      it 'resonds with a 302' do
+        get :new
+        expect(response.status).to eq(302)
+      end
+
+      it 'redirects to the login page' do
+        get :new
+        expect(response).to redirect_to '/login'
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with correct params' do
+
+      let(:article) { create(:article) }
+
+      before(:each) do
+        allow(controller).to receive(:current_user).and_return(article.user)
+      end
+
+      let(:params) { {article: {title: article.title,
```

Comment:
Could probably move this assignment just below `let(:article) {...`. Just to keep all variable assignments grouped together.